src/System.ComponentModel.TypeConverter/src/System/ComponentModel/GuidConverter.cs

```diff
@@ -1,86 +1,87 @@
-//------------------------------------------------------------------------------
-// <copyright file="GuidConverter.cs" company="Microsoft">
-//     Copyright (c) Microsoft Corporation.  All rights reserved.
-// </copyright>                                                                
-//------------------------------------------------------------------------------
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
 
-/*
- */
-namespace System.ComponentModel {
-    using Microsoft.Win32;
-    using System.ComponentModel.Design.Serialization;
-    using System.Diagnostics;
-    using System.Globalization;
-    using System.Reflection;
-    using System.Runtime.InteropServices;
-    using System.Runtime.Remoting;
-    using System.Runtime.Serialization.Formatters;
-    using System.Security.Permissions;    
+using System.Globalization;
 
+namespace System.ComponentModel
+{
     /// <devdoc>
     ///    <para>Provides a
     ///       type converter to convert globally unique identifier objects to and from various
     ///       other representations.</para>
     /// </devdoc>
-    [HostProtection(SharedState = true)]
-    public class GuidConverter : TypeConverter {
-
+    public class GuidConverter : TypeConverter
+    {
         /// <devdoc>
         ///    <para>Gets a value indicating whether this
         ///       converter can convert an object in the given source type to a globally unique identifier object
         ///       using the context.</para>
         /// </devdoc>
-        public override bool CanConvertFrom(ITypeDescriptorContext context, Type sourceType) {
-            if (sourceType == typeof(string)) {
+        public override bool CanConvertFrom(ITypeDescriptorContext context, Type sourceType)
+        {
+            if (sourceType == typeof(string))
+            {
                 return true;
             }
             return base.CanConvertFrom(context, sourceType);
         }
 
         /// <devdoc>
-        ///    <para>Gets a value indicating whether this converter can
-        ///       convert an object to the given destination type using the context.</para>
+        ///    <para>
+        ///        Gets a value indicating whether this converter can convert an object to
+        ///        the given destination type using the context.
+        ///    </para>
         /// </devdoc>
-        public override bool CanConvertTo(ITypeDescriptorContext context, Type destinationType) {
-            if (destinationType == typeof(InstanceDescriptor)) {
+        public override bool CanConvertTo(ITypeDescriptorContext context, Type destinationType)
+        {
+#if FEATURE_INSTANCEDESCRIPTOR
```

Comment:
`InstanceDescriptor` went behind a feature flag for now because typically it's used only like this, as a `CanConvertTo` special case.